### PR TITLE
console: display line number that error originated from in GraphQL editor (close #4849)

### LINKED ISSUE #4849: Show line number of the error in custom types editor
Currently the error message shows the syntax error. Line number would really be useful if the custom types are a lot

Screen Capture_select-area_20200522195437

### PR DESCRIPTION
**Pull request by tanvibhakta**
display line number for the first error encountered in the custom types editor

### Changelog

- [x] `CHANGELOG.md` is updated with user-facing content relevant to this PR. If no changelog is required, then add the `no-changelog-required` label.

### Affected components

- [x] Console

### Solution and Design

The error object available here is an object of type GraphQLError. It contains the location of the origin of the error within itself.

### Steps to test and verify
console > Actions tab > `Custom Types` button on the left hand side -> define a custom type in the editor.  Where previously an error used to show up, it now shows up with the line number the error is caught on as well.

### Limitations, known bugs & workarounds
The editor displays the first error message encountered, and therefore the line number displayed will be for that error.

### Related issues
https://github.com/hasura/graphql-engine/issues/4849

### CODE REVIEW

File: console/src/components/Services/Actions/Common/components/TypeDefinitionEditor.js [hunk 1]
```diff
@@ -52,6 +52,10 @@ const ActionDefinitionEditor = ({
 
   const errorMessage =
     error && (error.message || 'This is not valid GraphQL SDL');
+  const errorMessageLine =
+    error &&
+    error.locations &&
```
Comment (beerose) [inline console/src/components/Services/Actions/Common/components/TypeDefinitionEditor.js:57]: According to the type definition of `GraphQLError` `locations` can be an empty array, so let's check for `error.locations.length` here as well.

Comment (tanvibhakta) [reply console/src/components/Services/Actions/Common/components/TypeDefinitionEditor.js:57]: Oooh thanks for that!

**Comment by beerose** [comment]
Hi @tanvibhakta, would you mind doing the same thing for [Action Definition editor](https://github.com/hasura/graphql-engine/blob/b583bd8fce85dd932905c5f0e8ab693bf901b6e4/console/src/components/Services/Actions/Common/components/ActionDefinitionEditor.js)?

File: console/src/components/Services/Actions/Add/Add.js [hunk 2]
```diff
@@ -49,7 +52,7 @@ const AddAction = ({
     return () => {
       dispatch(resetDerivedActionParentOperation());
     };
-  }, []);
+  });
```
Comment (beerose) [inline console/src/components/Services/Actions/Add/Add.js:55]:
```suggestion
  }, []);
```

File: console/src/components/Services/Actions/Common/components/GraphQLEditor.tsx [hunk 3]
```diff
@@ -1,65 +1,86 @@
...
+import { Nullable } from '../../../../Common/utils/tsUtils';
 
-const editorLabel = 'Action definition';
-const editorTooltip =
-  'Define the action as a query or a mutation using GraphQL SDL. You can use the custom types already defined by you or define new types in the new types definition editor below.';
+type LocationError = {
+  line: number;
+  column: number;
+};
+interface GraphQLEditorError extends Error {
```
Comment (beerose) [inline console/src/components/Services/Actions/Common/components/GraphQLEditor.tsx:13]: Use `GraphQLError` from 'graphql' instead

File: CHANGELOG.md [hunk 4]
```diff
@@ -26,6 +26,7 @@
 - console: support tracking partitioned tables (close #5071) (#5258)
 - console: add button to cancel one-off scheduled events and cron-trigger events (close #5161) (#5236)
 - console: handle generated and identity columns in console data section (close #4552, #4863) (#4761)
+- console: display line number that error message originated from in custom types editor (#4849)
```
Comment (beerose) [inline CHANGELOG.md:29]:
```suggestion
- console: display line number that error originated from in GraphQL editor (close #4849) (#4942)
```

**Pull request merged**